app/code/Magento/Inventory/Model/StockSourceLink/Command/SaveStockSourceLinks.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Inventory\Model\StockSourceLink\Command;
+
+use Magento\Framework\Exception\CouldNotSaveException;
+use Magento\Framework\Validation\ValidationException;
+use Magento\Inventory\Model\ResourceModel\StockSourceLink as StockSourceLinkResourceModel;
+use Magento\InventoryApi\Api\Data\StockSourceLinkInterface;
+use Magento\InventoryApi\Api\StockSourceLinksSaveInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @inheritdoc
+ */
+class SaveStockSourceLinks implements StockSourceLinksSaveInterface
+{
+    /**
+     * @var StockSourceLinkResourceModel
+     */
+    private $stockSourceLinkResource;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param StockSourceLinkResourceModel $stockSourceLinkResource
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        StockSourceLinkResourceModel $stockSourceLinkResource,
+        LoggerInterface $logger
+    ) {
+        $this->stockSourceLinkResource = $stockSourceLinkResource;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(array $links): array
+    {
+        $savedLinkIdList = [];
+
+        try {
+            foreach ($links as $link) {
+                $this->stockSourceLinkResource->save($link);
```

Comment:
We could introduce multiple delete on Resource layer
Example in https://github.com/magento-engcom/msi/blob/develop/app/code/Magento/Inventory/Model/ResourceModel/SourceItem/SaveMultiple.php

Reply:
No problem, but this will change a way `Magento\Inventory\Controller\Adminhtml\Stock\StockSourceLinkProcessor` class assigned links. I will change that class and related stuff accordingly